platform-tests/src/test/java/org/junit/platform/engine/support/descriptor/ResourceLockAnnotationTests.java

```diff
@@ -346,51 +451,85 @@ public Set<Lock> provideForNestedClass(Class<?> testClass) {
 	}
 
 	@SuppressWarnings("JUnitMalformedDeclaration")
-	@ResourceLock
-	static class EmptyAnnotationTestCase {
+	@ResourceLock( //
+			value = "a1", //
+			target = ResourceLockTarget.CHILDREN, //
+			providers = SharedResourcesHavingTheSameValueAndModeAreDeduplicatedTestCase.Provider.class //
+	)
+	static class SharedResourcesHavingTheSameValueAndModeAreDeduplicatedTestCase {
 
 		@Test
-		@ResourceLock
+		@ResourceLock(value = "a1")
 		void test() {
 		}
 
-		@Nested
-		@ResourceLock
-		class NestedClass {
+		static class Provider implements ResourceLocksProvider {
+
+			@Override
+			public Set<Lock> provideForMethod(Class<?> testClass, Method testMethod) {
+				return Set.of(new Lock("a1"));
+			}
 		}
 	}
 
-	static class NestedNestedTestCase {
```

Comment:
Forgot to delete this test case draft in the previous [PR](https://github.com/junit-team/junit5/pull/3889) 😓